lua/lsp/init.lua

```diff
@@ -115,41 +82,51 @@ function M.common_on_attach(client, bufnr)
   end
   lsp_highlight_document(client)
   add_lsp_buffer_keybindings(bufnr)
-  require("lsp.null-ls").setup(vim.bo.filetype)
 end
 
-function M.setup(lang)
-  local lsp_utils = require "lsp.utils"
-  local lsp = lvim.lang[lang].lsp
-  if (lsp.active ~= nil and not lsp.active) or lsp_utils.is_client_active(lsp.provider) then
+function M.setup_by_ft(lang, opts)
+  vim.validate {
+    lang = { lang, "string" },
+    opts = { opts, "table" },
+    provider = { opts.lsp.provider, "string" },
+  }
+
+  if lsp_utils.is_client_active(opts.lsp.provider) then
     return
   end
 
+  if not opts.lsp.setup.on_attach then
+    opts.lsp.setup.on_attach = M.common_on_attach
+  end
+  if not opts.lsp.setup.on_init then
+    opts.lsp.setup.on_init = M.common_on_init
+  end
+  if not opts.lsp.setup.capabilities then
+    opts.lsp.setup.capabilities = M.common_capabilities()
+  end
+
+  lspconfig[opts.lsp.provider].setup(opts.lsp.setup)
+end
+
+local function bootstrap_nlsp()
+  local lsp_settings_status_ok, lsp_settings = pcall(require, "nlspsettings")
+  if lsp_settings_status_ok then
+    lsp_settings.setup {
+      config_home = utils.join_paths(get_config_dir(), "lsp-settings"),
+    }
+  end
+end
+
+local function is_overrided(lang)
   local overrides = lvim.lsp.override
```

Comment:
I haven't seen it used, but it feels like it will turn off the autosetup of the language-server. It should be removable if we end up going with some variant of `ensure_configured`. Since you can just exclude any language that you may want to setup manually.